src/Microsoft.AspNetCore.Localization/CookieRequestCultureProvider.cs

```diff
@@ -36,6 +42,8 @@ public override Task<ProviderCultureResult> DetermineProviderCultureResult(HttpC
                 throw new ArgumentNullException(nameof(httpContext));
             }
 
+            _logger = httpContext.RequestServices.GetService<ILogger>() ?? NullLogger.Instance;
```

Comment:
If I understand you correctly there should be two constructors, Am I right? If yes .. why we can make it one while the `CookieRequestCultureProvider` new up inside the `RequestLocalizationOptions`?